packages/engine/lib/execution/src/package/simulation/context/agent_messages/mod.rs

```diff
@@ -117,18 +119,34 @@ impl ContextPackage for AgentMessages {
         num_agents: usize,
         _schema: &ContextSchema,
     ) -> Result<Vec<(RootFieldKey, Arc<dyn Array>)>> {
-        let index_builder = ArrowIndexBuilder::new(1024);
-        let loc_builder = FixedSizeListBuilder::new(index_builder, 3);
-        let mut messages_builder = ListBuilder::new(loc_builder);
-
-        (0..num_agents).try_for_each(|_| messages_builder.append(true))?;
+        let index_builder = MutablePrimitiveArray::<u32>::with_capacity(1024);
+        let loc_builder = MutableFixedSizeListArray::new(index_builder, 3);
+        let mut messages_builder: MutableListArray<
+            i32,
+            MutableFixedSizeListArray<MutablePrimitiveArray<u32>>,
+        > = MutableListArray::new_from(
+            loc_builder,
+            DataType::List(Box::new(Field::new(
+                "item",
+                DataType::FixedSizeList(Box::new(Field::new("item", DataType::UInt32, true)), 3),
+                true,
+            ))),
```

Comment:
In `arrow-rs`, `"item"` was hardcoded. Is this still the case with `arrow2`? If no, we probably want to name it properly (can be done in a follow-up).